docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+forwarding proxy (i.e. the source IP is not preserved).  It is not trivial for
+kube-proxy to recognize the traffic therefore.  So, LoadBalancer services are
+also exposed as NodePort services.  For NodePort services, a cluster-wide port
+is assigned by kubernetes to the service, and kube-proxy listens externally on
+that port on every minion, and forwards traffic to the pods.  So for a
+load-balanced service, ELB is configured to proxy traffic on the public port
+(e.g. port 80) to the NodePort assigned to the service (e.g. 31234), kube-proxy
+recognizes the traffic coming to the NodePort by the inbound port number, and
+send it to the correct pods for the service.
+
+Note that we do not automatically open NodePort services in the AWS firewall
+(although we do open LoadBalancer services).  This is because we expect that
+NodePort services are more of a building block for things like inter-cluster
+services or for LoadBalancer.  To consume a NodePort service externally, you
+will likely have to open the port in the minion security group
+(`kubernetes-minion-<clusterid>`).
+
+### IAM
+
+kube-proxy sets up two IAM roles, one for the master called
+(kubernetes-master)[cluster/aws/templates/iam/kubernetes-master-policy.json]
+and one for the minions called
+(kubernetes-minion)[cluster/aws/templates/iam/kubernetes-minion-policy.json].
+
+The master is responsible for creating ELBs and configuring them, as well as
+setting up advanced VPC routing.  Currently it has blanket permissions on EC2,
+along with rights to create and destroy ELBs.
+
+The minion does not need a lot of access to the AWS APIs.  It needs to download
+a distribution file, and then it is responsible for attaching and detaching EBS
+volumes to itself.
+
+The minion policy is relatively minimal.  The master policy is probably overly
+permissive.  The security concious may want to lock-down the IAM policies
+further (#11936)
+
+We should make it easier to extend IAM permissions and also ensure that they
+are correctly configured (#???)
+
+### Tagging
+
+All AWS resources are tagged with a tag named "KuberentesCluster".  This tag is
+used to identify a particular 'instance' of Kubernetes, even if two clusters
+are deployed into the same VPC.  (The script doesn't do this by default, but it
+can be done.)
+
+Within the AWS cloud provider logic, we filter requests to the AWS APIs to
+match resources with our cluster tag.  So we only see our own AWS objects.
+
+If you choose not to use kube-up, you must tag everything with a
+KubernetesCluster tag with a unique per-cluster value.
+
+
+# AWS Objects
+
+The kube-up script does a number of things in AWS:
+
+* Creates an S3 bucket (`AWS_S3_BUCKET`) and copy the kubernetes distribution
+  and the salt scripts into it.  They are made world-readable and the HTTP URLs
+are passed to instances; this is how kubernetes code gets onto the machines.
+* Creates two IAM profiles based on templates in `cluster/aws/templates/iam`.
```

Comment:
replace the period with a colon and then list the two templates
.... `cluster/aws/templates/iam`:

     * `kubernetes-master`: Used by the Kubernetes master. 
     * `kubernetes-minion`: Used by the nodes.